File: prime/src/main/java/com/telenordigital/prime/config/PrimeConfiguration.java
package com.telenordigital.prime.config;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.dropwizard.Configuration;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

public final class PrimeConfiguration extends Configuration {

    @Valid
    @NotNull
    @JsonProperty("eventProcessor")
    private final EventProcessorConfiguration eventProcessor = new EventProcessorConfiguration();

    public EventProcessorConfiguration getEventProcessorConfig() {
        return eventProcessor;
    }
}

File: prime/src/main/kotlin/org/ostelco/prime/config/PrimeConfiguration.kt
package org.ostelco.prime.config

import com.fasterxml.jackson.annotation.JsonProperty
import io.dropwizard.Configuration

import javax.validation.Valid
import javax.validation.constraints.NotNull

class PrimeConfiguration : Configuration() {

    @Valid
    @NotNull
    @JsonProperty("eventProcessor")
    lateinit var eventProcessorConfig: EventProcessorConfiguration

    @Valid
    @NotNull
    @JsonProperty("ocs")
    lateinit var ocsConfiguration: OcsConfig
}
